### PR TITLE
Release 3.1.0

### LINKED ISSUE #117: Update RabbitMQ.Client to 3.6.1
Heads up , quite a few breaking changes - http://apicomparer.particular.net/compare/rabbitmq.client/3.5.0...3.6.0

### PR DESCRIPTION
**Pull request by andreasohlund**
Closes #117

### CODE REVIEW

**Comment by andreasohlund** [comment]
@Particular/transport-maintainers please review, ok with releasing this as a minor?

**Comment by SimonCropp** [comment]
"ok with releasing this as a minor"

yes

**Comment by bording** [comment]
One thing I'm curious about is how this will interact with the stuff that is on develop now. All of these changes have already been made there.

Once this is merged into master, what is the expectation for these commits in regard to develop?

**Comment by bording** [comment]
Also, do we want to include a fix for #122 in this release?

**Comment by mauroservienti** [comment]
In light of `safe by default`, IMO, yes we should fix #122

**Comment by andreasohlund** [comment]
Yea, lets include #122

**Comment by andreasohlund** [comment]
>One thing I'm curious about is how this will interact with the stuff that is on develop now. All of these changes have already been made there.

>Once this is merged into master, what is the expectation for these commits in regard to develop?

I guess we'll find out once that happens :)

**Comment by bording** [comment]
A fix for #122 has been pushed to the branch.

**Comment by andreasohlund** [comment]
Seems like it broke 2 tests?

File: src/NServiceBus.RabbitMQ/RabbitMqDequeueStrategy.cs [hunk 1]
```diff
@@ -156,7 +154,7 @@ void ConsumeMessages(object state)
 
                     var consumer = new QueueingBasicConsumer(channel);
 
-                    channel.BasicConsume(parameters.Queue, noAck,receiveOptions.ConsumerTag, consumer);
+                    channel.BasicConsume(parameters.Queue, false, receiveOptions.ConsumerTag, consumer);
```
Comment (andreasohlund) [inline src/NServiceBus.RabbitMQ/RabbitMqDequeueStrategy.cs]: Since tx == off means "fire and forget" we need to ack right away "if txmode == none" ?

**Comment by bording** [comment]
I pushed up a revised commit that should address the tests that were failing. I didn't realize that the transport was responsible for requeuing messages in v5 for FLR.

I revised it to look at the `IsTransactional` flag to determine if the `BasicReject` calls should requeue the message or not.

This should provide the desired behavior in the majority of cases. The only time I can think where it would be different would be if the Rabbit connection closed before `BasicReject` could be called. In that case, the message would be requeued since it was not acked or rejected, even if transactions are disabled.

Is that ok? If not, I can revise it further.

**Comment by SimonCropp** [comment]
A new rabbitmq clent was released last night. Does that effect anyhting here?

**Comment by bording** [comment]
I was just about to push that up too. If we're updating the client, makes sense to me to go for the latest.

**Comment by bording** [comment]
Just pushed up another revision to the autoack fix. Now, if `IsTransactional` is false, `BasicAck()` gets called before we try to process the message.

This makes it much less likely that a connection interruption could cause the message to be requeued when we don't want it to be. I'm pretty sure there is still a potential race between calling `BasicAck()` and the client knowing that a channel has been closed, but I'm not sure there is anything we can do about that.

**Comment by andreasohlund** [comment]
Looks good, can you go ahead and release this @bording ?

**Comment by bording** [comment]
3.1.0 has been released and announced on GG.

**Pull request merged**